src/ndpp_ops.F90

```diff
@@ -0,0 +1,1083 @@
+          this % el(iE) % outgoing(:, gmin : gmax) = &
+            temp_outgoing(1 : scatt_order, gmin : gmax)
+          deallocate(temp_outgoing)
+        end if
+      end do
+
+      ! The remainder only apply to nuclides (inelastic, nu-inelastic and chi data)
+      if (is_nuc) then
+        read(UNIT=in, FMT='(I20)') NEin
+        if (NEin > 0) then
+          ! Now allocate all that will be filled
+          allocate(this % inel_Ein(NEin))
+          allocate(this % inel_Ein_srch(NG + 1))
+          allocate(this % inel(NEin))
+
+          ! Now read in inel_Ein, inel_Ein_srch
+          read(UNIT=in, FMT=*) this % inel_Ein
+          read(UNIT=in, FMT=*) this % inel_Ein_srch
+
+          ! Get the inelastic moments themselves
+          do iE = 1, NEin
+            ! get gmin and gmax
+            read(UNIT=in, FMT='(I20,I20)') gmin, gmax
+
+            if ((gmin > 0) .and. (gmax > 0)) then
+              ! Then we can allocate the space. Do it to lib_order
+              ! since this is the largest order requested in the tallies.
+              ! Since we only need to store up to the maximum, we also need to have
+              ! an array for reading the file which we can later truncate to fit
+              ! in to this % inel(iE) % outgoing.
+              allocate(temp_outgoing(lib_order, gmin : gmax))
+
+              ! Now we have a space to store the data, get it.
+              read(UNIT=in, FMT=*) temp_outgoing
+              ! And copy in to this % el
+              allocate(this % inel(iE) % outgoing(scatt_order, gmin : gmax))
+              this % inel(iE) % outgoing(:, gmin : gmax) = &
+                temp_outgoing(1 : scatt_order, gmin : gmax)
+              deallocate(temp_outgoing)
+            end if
+          end do
+
+          ! Get nu-scatter, if needed
+          if (nuscatter == 1) then
+            allocate(this % nuinel(NEin))
+            do iE = 1, NEin
+              ! get gmin and gmax
+              read(UNIT=in, FMT='(I20,I20)') gmin, gmax
+
+              if ((gmin > 0) .and. (gmax > 0)) then
+                ! Then we can allocate the space. Do it to lib_order
+                ! since this is the largest order requested in the tallies.
+                ! Since we only need to store up to the maximum, we also need to have
+                ! an array for reading the file which we can later truncate to fit
+                ! in to this % nuinel(iE) % outgoing.
+                allocate(temp_outgoing(lib_order, gmin : gmax))
+
+                ! Now we have a space to store the data, get it.
+                read(UNIT=in, FMT=*) temp_outgoing
+                allocate(this % nuinel(iE) % outgoing(scatt_order, &
+                  gmin : gmax))
+                this % nuinel(iE) % outgoing(:, gmin : gmax) = &
+                  temp_outgoing(1 : scatt_order, gmin : gmax)
+                deallocate(temp_outgoing)
+              end if
+            end do
+          end if
+        end if
```

Comment:
This code is nested pretty deep.  It would help readability to put some labels on some of the outermost loops.